pandas/tests/groupby/test_function.py

```diff
@@ -1087,3 +1087,11 @@ def test_groupby_sum_below_mincount_nullable_integer():
     result = grouped.sum(min_count=2)
     expected = DataFrame({"b": [pd.NA] * 3, "c": [pd.NA] * 3}, dtype="Int64", index=idx)
     tm.assert_frame_equal(result, expected)
+
+
+def test_groupby_index_object_dtype():
+    # GH 40014
+    df = DataFrame({"c0": ["x", "x", "x"], "c1": ["x", "x", "y"], "p": [0, 1, 2]})
+    df.index = df.index.astype("O")
+    grouped = df.groupby(["c0", "c1"])
+    grouped.p.agg(lambda x: all(x > 0))
```

Comment:
Even though this test does verify no error is raised, may as well also follow result/expected pattern and verify correct output

Reply:
Done!